snare/tests/test_replace_links.py

```diff
@@ -0,0 +1,72 @@
+import unittest
+import sys
+import os
+import shutil
+import asyncio
+from snare.cloner import Cloner
+from snare.utils.page_path_generator import generate_unique_path
+
+
+class TestReplaceLinks(unittest.TestCase):
+    def setUp(self):
+        self.main_page_path = generate_unique_path()
+        os.makedirs(self.main_page_path)
+        self.root = 'http://example.com'
+        self.level = 0
+        self.max_depth = sys.maxsize
+        self.loop = asyncio.new_event_loop()
+        self.css_validate = 'false'
+        self.handler = Cloner(self.root, self.max_depth, self.css_validate)
+        self.content = '''
+                          <html>
+                                <body>
+                                      <a href="http://example.com/test"></a>
+                                </body>
+                          </html>
+                       '''
+        self.expected_content = None
+        self.return_content = None
+
```

Comment:
Check whether `process_link` is called with certain parameters or not.

Reply:
What you are doing here is basically the test cases for `process_link` method so add them to the above file. And for this method just check whether `process_link` is called or not with required params and mock the `process_link ` method.

Reply:
`process_link` already has such type of test cases but i will change this implementation with `assert_called_with`.

Reply:
You didn't check `process_links` for an absolute URL.

Reply:
Now Checked!